src/include/platform/internal/GenericConfigurationManagerImpl.ipp

```diff
@@ -423,7 +423,18 @@ void GenericConfigurationManagerImpl<ImplClass>::NotifyOfAdvertisementStart()
 template <class ConfigClass>
 CHIP_ERROR GenericConfigurationManagerImpl<ConfigClass>::GetRegulatoryLocation(uint8_t & location)
 {
-    return GetLocationCapability(location);
+    uint32_t value;
+    if (CHIP_NO_ERROR != ReadConfigValue(ConfigClass::kConfigKey_RegulatoryLocation, value))
+    {
+        ReturnErrorOnFailure(GetLocationCapability(location));
+        ReturnErrorOnFailure(StoreRegulatoryLocation(location));
```

Comment:
The spec says LocationCapability keeps the default value of RegulatoryLocation, if failed to read RegulatoryLocation, we can read the value from LocationCapability instead, but the StoreRegulatoryLocation does not need to be success in this case.

Reply:
Fixed